src/lustre/lustreSyntaxChecks.ml

```diff
@@ -91,6 +98,7 @@ let error_message kind = match kind with
   | UnsupportedAssignment -> "Assignment not supported"
   | AssumptionVariablesInContractNode -> "Assumption variables not supported in contract nodes"
   | ClockMismatchInMerge -> "Clock mismatch for argument of merge"
+  | MisplacedVarInFrameBlock id -> "Misplaced var '" ^ HString.string_of_hstring id ^ "' is not declared in the frame block header"
```

Comment:
I would replace this message with the following message:
`Variable 'x' is defined in the frame block but not declared in the frame block header`